precompiles/gov/integration_test.go

```diff
@@ -475,5 +477,202 @@ var _ = Describe("Calling governance precompile from EOA", func() {
+					uint32(0), // StatusNil to get all proposals
+					common.Address{},
+					common.Address{},
+					query.PageRequest{
+						Limit:      10,
+						CountTotal: true,
+					},
+				}
+				txArgs.GasLimit = 200_000
+
+				_, ethRes, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					callArgs,
+					passCheck,
+				)
+				Expect(err).To(BeNil(), "error while calling the smart contract: %v", err)
+
+				var out gov.ProposalsOutput
+				err = s.precompile.UnpackIntoInterface(&out, method, ethRes.Ret)
+				Expect(err).To(BeNil())
+
+				Expect(out.Proposals).To(HaveLen(1))
+				Expect(out.PageResponse.Total).To(Equal(uint64(1)))
+
+				proposal := out.Proposals[0]
+				Expect(proposal.Id).To(Equal(uint64(1)))
+				Expect(proposal.Status).To(Equal(uint32(v1.StatusVotingPeriod)))
+				Expect(proposal.Proposer).To(Equal(s.keyring.GetAddr(0)))
+				Expect(proposal.Messages).To(HaveLen(1))
+				Expect(proposal.Messages[0]).To(Equal("/cosmos.bank.v1beta1.MsgSend"))
+			})
+
+			It("should filter proposals by status", func() {
+				callArgs.Args = []interface{}{
+					uint32(v1.StatusVotingPeriod),
+					common.Address{},
+					common.Address{},
+					query.PageRequest{
+						Limit:      10,
+						CountTotal: true,
+					},
+				}
+				txArgs.GasLimit = 200_000
+
+				_, ethRes, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					callArgs,
+					passCheck,
+				)
+				Expect(err).To(BeNil())
+
+				var out gov.ProposalsOutput
+				err = s.precompile.UnpackIntoInterface(&out, method, ethRes.Ret)
+				Expect(err).To(BeNil())
+
+				Expect(out.Proposals).To(HaveLen(1))
+				Expect(out.Proposals[0].Status).To(Equal(uint32(v1.StatusVotingPeriod)))
+			})
+
+			It("should filter proposals by voter", func() {
+				// First add a vote
+				voteArgs := factory.CallArgs{
+					ContractABI: s.precompile.ABI,
+					MethodName:  gov.VoteMethod,
+					Args: []interface{}{
+						s.keyring.GetAddr(0), uint64(1), uint8(v1.OptionYes), "",
+					},
+				}
+				_, _, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					voteArgs,
+					passCheck.WithExpEvents(gov.EventTypeVote),
+				)
+				Expect(err).To(BeNil())
+
+				// Wait for the vote to be included in the block
+				Expect(s.network.NextBlock()).To(BeNil())
+
+				// Query proposals filtered by voter
+				callArgs.Args = []interface{}{
+					uint32(0), // StatusNil
+					s.keyring.GetAddr(0),
+					common.Address{},
+					query.PageRequest{
+						Limit:      10,
+						CountTotal: true,
+					},
+				}
+				txArgs.GasLimit = 200_000
+
+				_, ethRes, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					callArgs,
+					passCheck,
+				)
+				Expect(err).To(BeNil())
+
+				var out gov.ProposalsOutput
+				err = s.precompile.UnpackIntoInterface(&out, method, ethRes.Ret)
+				Expect(err).To(BeNil())
+
+				Expect(out.Proposals).To(HaveLen(1))
```

Comment:
we should probably set up multiple proposals and then only have the one for the voter returned and then check if it's the expected prop. if there's only one in total we can't be sure it's filtered correctly